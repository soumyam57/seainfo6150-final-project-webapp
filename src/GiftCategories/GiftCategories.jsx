import React, { useEffect, useState } from 'react';
import wedding from "../images/wedding.jpg";
import christmas from "../images/christmas.jpg";
import dolls from "../images/dolls.jpg";
import Birthday from "../images/Birthday.jpg";
import getwell from "../images/getwell.jpg";
import styles from "./GiftCategories.module.css";
import { Link } from "react-router-dom";

//http://demo5934733.mockable.io/

const GiftCategories = () => {
    return(
            <div className={styles.container}>
                <div className={styles.flex}>
                    <img className={styles.image} src={wedding} alt=""/>
                    <p className={styles.text}>   
                        <Link className={styles.text} to='/wedding'>
                            Wedding & Party
                        </Link>
                    </p> 
                </div>

                <div className={styles.flex}>
                    <img className={styles.image} src={christmas} alt=""/>
                    <p className={styles.text}>
                        <Link className={styles.text} to='/festival'>
                            Holidays & Festival
                        </Link>
                    </p>
                    
                </div>
                
                <div className={styles.flex}>
                    <img className={styles.image} src={dolls} alt="" />
                    <p className={styles.text}> 
                        <Link className={styles.text} to='/kids'>
                            Kids
                        </Link>
                    </p>
                </div>

                <div className={styles.flex}>
                    <img className={styles.image} src={Birthday} alt="" />
                    <p className={styles.text}> 
                        <Link className={styles.text} to='/birthday'>
                            Birthday
                        </Link>
                    </p>
                </div>

                <div className={styles.flex}>
                    <img className={styles.image} src={getwell} alt=""/>  
                    <p className={styles.text}>
                        <Link className={styles.text} to='/getwell'>
                            Get Well Soon
                        </Link>
                    </p>
                </div>
        </div>
    )
}


export default GiftCategories;


/*



    */
