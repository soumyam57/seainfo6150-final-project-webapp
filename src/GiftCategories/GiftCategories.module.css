.container {
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    padding: 0px 10px 50px 5px;
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
    border: 0px solid #666;
    display: grid;
    grid-template-columns: 1fr;
    align-items: center;
  }
  
  @media screen and (min-width: 525px) {
    .container {
      grid-template-columns: repeat(2, 1fr);
    }
  }
  
  @media screen and (min-width: 800px) {
    .container {
      grid-template-columns: repeat(3, 1fr);
    }
  }
  
  @media screen and (min-width: 900px) {
    .container {
      grid-template-columns: repeat(5, 1fr);
    }
  }

.flex {
    align-items: center;
    height: 50%;
    display: flex;
    flex-direction: column;  
}

.image{
    padding: 10px;
    width:60%;
    height: 100%;
    border-radius: 50%;
    box-shadow:0px 0px 10px 5px grey;
    border: 5px solid maroon;
}

.text {
  padding: 20px 0px 10px 0px;
    justify-content: center;
    color: gainsboro;
    font-style: italic;
    font-weight: bolder;
    text-transform: capitalize;
    text-decoration-line: none;
}

.title {
    background-color: gainsboro;
    padding: 10px 50px;
    color: maroon;
    font-style: italic;
    font-weight: bold;
    text-transform: uppercase;
}