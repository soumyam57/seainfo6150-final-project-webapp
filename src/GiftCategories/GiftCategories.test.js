import React from "react";
import { render } from "@testing-library/react";
import GiftCategories from "./GiftCategories.jsx";
import { BrowserRouter } from 'react-router-dom';

it("renders the GiftCategories component correctly", () => {
  const { container } = render(
    <BrowserRouter>  
        <GiftCategories/>
    </BrowserRouter> 
  );
  expect(container).toMatchSnapshot();
});