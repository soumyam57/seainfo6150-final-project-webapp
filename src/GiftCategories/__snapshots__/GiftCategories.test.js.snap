// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the GiftCategories component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="wedding.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/wedding"
        >
          Wedding & Party
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="christmas.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/festival"
        >
          Holidays & Festival
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="dolls.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/kids"
        >
          Kids
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="Birthday.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/birthday"
        >
          Birthday
        </a>
      </p>
    </div>
    <div
      class="flex"
    >
      <img
        alt=""
        class="image"
        src="getwell.jpg"
      />
      <p
        class="text"
      >
        <a
          class="text"
          href="/getwell"
        >
          Get Well Soon
        </a>
      </p>
    </div>
  </div>
</div>
`;
